hub/src/server/server.js

```diff
@@ -69,6 +68,48 @@ export class HubServer {
       contentType = 'application/octet-stream'
     }
 
+
+    // can the caller write? if so, in what paths?
+    const scopes = getAuthenticationScopes(requestHeaders.authorization)
+    const writePrefixes = []
+    const writePaths = []
+    for (let i = 0; i < scopes.length; i++) {
+      if (scopes[i].scope == 'putFilePrefix') {
+        writePrefixes.push(scopes[i].domain)
+      } else if (scopes[i].scope == 'putFile') {
+        writePaths.push(scopes[i].domain)
+      }
+    }
+
+    if (writePrefixes.length > 0 || writePaths.length > 0) {
+      // we're limited to a set of prefixes and paths.
+      // does the given path match any prefixes?
+      let match = false
+      if (writePrefixes.length > 0) {
+        for (let i = 0; i < writePrefixes.length; i++) {
```

Comment:
Actually, same advice re: find applies here. I think using `.find` over loop/breaks improves the code clarity --- however, if you feel strongly otherwise, that's okay with me.